### DISCUSSION TITLE
sql query update auth.users schema

### DISCUSSION THREAD

**Discussion by ajgil** [supabase/supabase#7836] [Questions] [answered]
Some days ago it worked but i dont know why today doesnt.

I wanna update some data from sql before do it on my app.

![Screenshot 2022-07-21 at 16 28 12](https://user-images.githubusercontent.com/29748376/180241909-8a4a92a2-1db5-4d48-af55-663e49fad23c.png)
![Screenshot 2022-07-21 at 16 40 58](https://user-images.githubusercontent.com/29748376/180242539-21a65c5a-0236-447f-96ca-f9bf06031f72.png)

Is there some wrong?

**Comment by GaryAustin1** [comment]
Do you have other code you are not showing that might be causing the error?

**Comment by ajgil** [comment]
the error is :
`Failed to run sql query: missing FROM-clause entry for table "users"`

the sql code:
```
UPDATE auth.users
    SET raw_user_meta_data = raw_user_meta_data || '{"ode": true, "verified": true}'
WHERE id = 'user uuid';
```

**Reply by GaryAustin1** [reply]
Do you have a trigger function on auth.users?
I don't get why it is complaining about FROM on an UPDATE call, hence question if there is other code running.

**Reply by silentworks** [reply]
@ajgil please copy the information from here and add to the original message you posted, you can edit the original message and add in the additional information from all the other replies you've made in this thread.

**Comment by ajgil** [comment]
I have 2 triggers and functions, 

first is about create public.profiles, modified from supabase example

second is about update public.profiles from that function trigger (not executed yet) cause need update twillio OTP account

first function trigger it found perfect
```sql
create or replace function public.handle_new_kluber() 
returns trigger as $$
begin
  
  INSERT INTO public.klubers ( id, avatar_url, full_name, email, ode, phone )
    VALUES ( 
      new.id, 
      new.raw_user_meta_data->>'avatar_url', 
      new.raw_user_meta_data->>'full_name', 
      new.email, 
      false,
      new.raw_user_meta_data->>'phone' );

  return new;
end;
$$ language plpgsql security definer;

create trigger on_auth_new_kluber
  after insert on auth.users
  for each row execute procedure public.handle_new_kluber();
```

second function trigger - not executed yet and guess it should work

```sql
create or replace function public.handle_new_ode() 
returns trigger as $$
begin

    update public.klubers k
    SET (k.ode, k.verified, k.phone, k.ode_date) = (true, true, NEW.phone, NEW.updated_at)
    where auth.users.id = k.id;

  return new;
end;
$$ language plpgsql security invoker;

create trigger on_auth_new_ode
  after update on auth.users
  for each row execute procedure public.handle_new_ode();
```

**Comment by ajgil** [comment]
So i want to update auth.users by sql statement and see if function `handle_new_ode()` works

**Comment by GaryAustin1** [comment, marked as answer]
OK we need to stop generating answers each time by replying instead of answering...  Please let's follow up in this answer...

When you run your SQL that is running any triggers on auth.users for update.
I don't see a select, or something that might look like a select that would cause a missing from error though in that function.

**Reply by bnjmnt4n** [reply]
Hey @ajgil! I believe the issue is in your `handle_new_ode` function. You're running the following:

```sql
    update public.klubers k
    SET (k.ode, k.verified, k.phone, k.ode_date) = (true, true, NEW.phone, NEW.updated_at)
    where auth.users.id = k.id;
```

However, in this `update` statement, where you're trying to update `public.klubers`, it isn't clear how to obtain a value from `auth.users` since you didn't specify `auth.users` in a `from` clause, which is what the error is complaining. Since you're using a trigger where you know the individual `auth.users` row being updated, I think something like the following should work:

```sql
    update public.klubers k
    SET (ode, verified, phone, ode_date) = (true, true, NEW.phone, NEW.updated_at)
    where k.id = NEW.id;
```

**Reply by ajgil** [reply]
Thanks @bnjmnt4n it works... what a mess hahaha  2 functions and triggers are working perfect.
